core/src/main/java/org/apache/druid/timeline/SegmentWithOvershadowedStatus.java

```diff
@@ -31,6 +32,12 @@
 public class SegmentWithOvershadowedStatus implements Comparable<SegmentWithOvershadowedStatus>
 {
   private final boolean overshadowed;
+  /**
+   * dataSegment is serialized "unwrapped", i.e. it's properties are included as properties of
+   * enclosing class. If in future, if {@Code SegmentWithOvershadowedStatus} were to extend {@link DataSegment},
+   * there will be no change in the serialized format.
+   */
+  @JsonUnwrapped
```

Comment:
Can you please also ensure that the data is sent in smile format, not JSON, between Coordinator and Brokers?

Reply:
created #7798